Storage/Storage/Bookmarks.swift

```diff
@@ -34,14 +36,11 @@ public struct BookmarkRoots {
 /**
  * The immutable base interface for bookmarks and folders.
  */
-@objc public protocol BookmarkNode {
-    var guid: String { get }
-    var title: String { get }
-    var icon: UIImage { get }
-}
-
-@objc public protocol Bookmark : BookmarkNode {
-    var url: String! { get }
+public class BookmarkNode {
+    public var id: Int? = nil
+    public var guid: String = ""
+    public var title: String = ""
```

Comment:
Why is `title` _not_ optional? Semanitcally, an empty string means it has a title, but that title is an empty string. `nil` means it has no title. Shouldn't we prefer the latter?